package.json

```diff
@@ -116,13 +116,11 @@
 		],
 		"rules": {
 			"@typescript-eslint/no-empty-function": "off",
-			"@typescript-eslint/no-base-to-string": "off",
 			"node/prefer-global/url": "off",
 			"node/prefer-global/url-search-params": "off",
-			"@typescript-eslint/prefer-readonly-parameter-types": "off",
+			"import/no-anonymous-default-export": "off",
 			"@typescript-eslint/no-floating-promises": "off",
```

Comment:
@szmarczak We need to enable this rule. It will help us catch unhandled exceptions errors. I think it's better if you do it though, as you can better answer whether something is safe to leave as floating or not. See https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md

Reply:
Well I've fixed it but it says that [this](https://github.com/sindresorhus/got/blob/b51d836f27b3dec16871ec3367d363607af42bb2/test/progress.ts#L15-L42) returns a Promise even though we clearly see that it doesn't.

I think we should revert this PR to 6b7e0ec and go with this option off. Maybe make an issue about this.

Reply:
It's really a pity you cannot mark a promise as `safe` (non-throwing). If it was, it should skip the error if there's a `return;` after the promise since it's an async task. Like some sort of events.

Reply:
> It's really a pity you cannot mark a promise as safe (non-throwing).

Such an annotation would be useless as anything can throw in JS. Even accessing a defined property can throw (proxies/getter).

The good thing about this `void` annotation is that it forces you to think about the problem and decide whether it's reasonably safe to keep the promise floating or not. It's usually not, per my comment that anything can potentially throw.

Reply:
> Well I've fixed it but it says that this returns a Promise even though we clearly see that it doesn't.

Just add an ESLint ignore comment on it. Something there are false-positives if a type is some kind of `any` as it could potentially be a Promise then, TS doesn't know.

Reply:
Travis is still failing.

Reply:
Actually it's the `checkEvents` function was improperly typed.